### ISSUE TITLE
MVM for CLIP feature

### ISSUE THREAD

**Issue by vateye** [tsujuifu/pytorch_violet#12]
Hi, I would like to know how to compute the loss between VideoSwin and the CLIP features in latest paper. Since the Swin family models take the patch size as 4x4, however for ViT the patch size is 16. I would like to know how to compute the loss between these two? (i.e., l1 loss)?

Thanks.

**Comment by tsujuifu** [comment]
Hi vateye, 

Thanks for your interest in our empirical study.

We use VidSwin and CLIP (ViT-B/32), both **with a patch size of 32**. 
For example, a 224<sup>2</sup> video frame will be 7<sup>2</sup> visual features.

**Comment by vateye** [comment]
Thanks! Meanwhile, I am curious how do you apply Random Masking and Blockwise Masking together with the ratio of 30%. Is there any code for doing this? Since these two are disjoint masking strategies.

**Comment by tsujuifu** [comment]
This code only has [BM](https://github.com/tsujuifu/pytorch_violet/blob/main/main_pretrain.py#L130) and [AM](https://github.com/tsujuifu/pytorch_violet/blob/main/main_pretrain.py#L145).
During pre-training, we [randomly chose](https://github.com/tsujuifu/pytorch_violet/blob/main/main_pretrain.py#L130) among them. You can also use a different probability.
